demo_net/ZeroDowntimeDeployment/Middlewares/LivenessProbeMiddleware.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using ZeroDowntimeDeployment.Services;
+
+namespace ZeroDowntimeDeployment.Middlewares
+{
+    public class LivenessProbeMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly IHealthService _healthService;
+
+        public LivenessProbeMiddleware(
+            RequestDelegate next,
+            IHealthService healthService)
+        {
+            _next = next;
+            _healthService = healthService;
+        }
+
+        public async Task InvokeAsync(HttpContext context)
+        {
+            var path = context.Request.Path.Value.ToLowerInvariant();
+            switch (path)
+            {
+                case "/probe":
+                // Support for Zero downtime deployment with kubernetes scenarios
+                case "/healthz":
```

Comment:
Please also return: OK <hostname> and NOT OK <hostname>.
So, during the demo, it is clear which instances of our demo are unhealthy.